### ISSUE TITLE
Greek/non-ASCII character input/handling problems

### ISSUE THREAD

**Issue by KubaO** [wxMaxima-developers/wxmaxima#1341]
This is a master issue to track this problem. The following other issues are open at the moment.

Bugs:

- [x] #1065 
- [x] #769
- [x] #725 
- [x] #710
- [x] #590

Feature requests:
- [x] #1336 Detect if lisp supports Unicode

I believe that the only choice we have is to investigate carefully any involved configuration, and coax wxMaxima into a behavior that works around limitations of given Maxima + lisp combination. I don't think we can punt this onto unsuspecting users, since it is a big part of user experience. I fully believe that we can work around these issues on wxMaxima side, with collaboration from `wxmathml.lisp`.

I've set a September milestone since this will take some effort.

**Comment by gunterkoenigsmann** [comment]
#590 is a completely different issues to the others: It was a change in how greek characters had to be specified to gnuplot: In the past a \greek{a} made an alpha. Nowadays gnuplot supports unicode.

**Comment by KubaO** [comment]
Thank you for clarifying. This was my attempt at bug triage - as you can see, with mixed results (also: I probably did it too late at night).

**Comment by gunterkoenigsmann** [comment]
No complaints from my side - and I am afraid you might have stepped into a trap I had set up for others:

A few years ago there was a discussion if maxima should provide complete unicode support or drop unicode support entirely. Url-encoding all variable-names was no alternative to suppporting unicode as maxima's file I/O commands and gnuplot interface still wouldn't have had unicode support, in this case and everything that tells maxima to convert equations to strings and vice versa would have been instantly broken in that case. As it looked like they would drop unicode support and as the discussion took place immediately before a release I therefore added the unicode sidebars to wxMaxima. The result was a userbase that didn't want to loose that feature. Also it was easy to test for unicode regressions - and both projects received bug tickets for the last remaining unicode bugs. 

The only bugs I completely failed to resolve back then were the ones in which the variable names changed the outcome of operations. In the meantime I have learned there are two classes of such bugs: Several maxima functions choke on using "x" or "y" as variable name. And for some functions the outcome depends on the alphabetical order of the variable names they work on. Fortunately none of these two is a unicode bug we have to fix ;-)
